LiveTV errors with TVHeadEnd

**Describe the bug**
I installed the plugin as explained [here](https://jellyfin.org/docs/general/clients/kodi.html) on my LibreElec 10.0.2 (on a RPI4) with Kodi v19.4 without issues. Setting the account wasn't a problem either.
For the moment all I have set in Jellyfin is the TVHeadEnd plugin (no other media than TV channels and recordings).
When I try to stream a channel (or watch a recording), I get an error on Kodi and nothing happens.
Streaming the same channels work from the Jellyfin Android app (installed from F-Droid) and from Firefox.

**To Reproduce**
1. Install the plugin
2. Configure it with server url / user / password
3. Try to stream any channel
4. See error

**Expected behavior**
The channels should start to stream as it happens on the Android app and on Firefox.

**Logs**
```
2022-08-26 19:28:16.315 T:1856     INFO <general>: CAddonMgr::FindAddon: plugin.video.jellycon v0.5.6+py3 installed
2022-08-26 19:28:16.334 T:1974     INFO <general>: initializing python engine.
2022-08-26 19:28:16.660 T:786   WARNING <general>: CGUIMediaWindow::OnMessage - updating in progress
2022-08-26 19:28:17.929 T:1974    ERROR <general>: GetDirectory - Error getting /storage/.kodi/userdata/addon_data/plugin.video.jellycon/
---
2022-08-26 19:33:40.989 T:2864     INFO <general>: JELLYFIN.resources.lib.functions -> INFO::resources/lib/functions.py:847 Sending jellycon_play_action : {'action': 'play', 'item_id': 'e80a
136a0d0d750360d6bd5f2bd9d0cf', 'auto_resume': '-1', 'force_transcode': False, 'media_source_id': '', 'subtitle_stream_index': None, 'audio_stream_index': None}
2022-08-26 19:33:41.071 T:2864    ERROR <general>: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                                    - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'KeyError'>
                                                   Error Contents: 'Container'
                                                   Traceback (most recent call last):
                                                     File "/storage/.kodi/addons/plugin.video.jellycon/default.py", line 18, in <module>
                                                       main_entry_point()
                                                     File "/storage/.kodi/addons/plugin.video.jellycon/resources/lib/tracking.py", line 23, in wrapper
                                                       value = func(*args, **kwargs)
                                                     File "/storage/.kodi/addons/plugin.video.jellycon/resources/lib/functions.py", line 146, in main_entry_point
                                                       play_action(params)
                                                     File "/storage/.kodi/addons/plugin.video.jellycon/resources/lib/functions.py", line 848, in play_action
                                                       play_file(play_info)
                                                     File "/storage/.kodi/addons/plugin.video.jellycon/resources/lib/tracking.py", line 23, in wrapper
                                                       value = func(*args, **kwargs)
                                                     File "/storage/.kodi/addons/plugin.video.jellycon/resources/lib/play_utils.py", line 384, in play_file
                                                       playurl, playback_type, listitem_props = get_play_url(selected_media_source, play_session_id, channel_id)
                                                     File "/storage/.kodi/addons/plugin.video.jellycon/resources/lib/play_utils.py", line 1146, in get_play_url
                                                       container = media_source["Container"]
                                                   KeyError: 'Container'
                                                   -->End of Python script error report<--
                                                   
2022-08-26 19:33:41.417 T:2864     INFO <general>: Python interpreter stopped
```

Just as an information, the folder `/storage/.kodi/userdata/addon_data/plugin.video.jellycon/` was present on the FS.

**System :**
 - OS: LibreElec 10.0.2 (RPI4)
 - Jellyfin Version: 10.8.4
 - Kodi Version: 19.4
 - Addon Version: 0.5.6+py3

Thanks for your reply. The debug logging is below.
PS : since yesterday I also uninstalled and reinstalled the plugin and I see now that Jellycon is in version 0.5.7+py3.
Also (maybe it is an important information), Jellyfin is installed in a docker container on my server (not the same hardware as my RPI4 which has LibreElec on it) using latest jellyfin/jellyfin image from dockerhub and having only the port 8096 mapped.
The line you asked for says : `INFO <general>: JELLYFIN.resources.lib.play_utils -> DEBUG::resources/lib/play_utils.py:1125 get_play_url - media_source: {0}`

```
2022-08-27 16:05:50.322 T:2783     INFO <general>: initializing python engine.
2022-08-27 16:05:51.098 T:2783     INFO <general>: JELLYFIN.default -> DEBUG::default.py:16 About to enter mainEntryPoint()
2022-08-27 16:05:51.099 T:2783     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:46 ===== JellyCon START =====
2022-08-27 16:05:51.105 T:2783     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:57 Running Python: sys.version_info(major=3, minor=8, micro=9, releaselevel='final', serial=0)
2022-08-27 16:05:51.105 T:2783     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:58 Running JellyCon: 0.5.7+py3
2022-08-27 16:05:51.106 T:2783     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:59 Kodi BuildVersion: 19.4 (19.4.0) Git:19.4-Matrix
2022-08-27 16:05:51.106 T:2783     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:60 Kodi Version: 19
2022-08-27 16:05:51.106 T:2783     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:61 Script argument data: ['plugin://plugin.video.jellycon/', '-1', '?item_id=1ed8d400xxxxxxxxxx40c0fc503991e9&mode=PLAY', 'resume:false']
2022-08-27 16:05:51.106 T:2783     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:317 Parameter string: ?item_id=1ed8d400xxxxxxxxxx40c0fc503991e9&mode=PLAY
2022-08-27 16:05:51.107 T:2783     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:318 Plugin Path string: plugin://plugin.video.jellycon/
2022-08-27 16:05:51.107 T:2783     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:326 JellyCon -> Detected parameters: {'item_id': '1ed8d400xxxxxxxxxx40c0fc503991e9', 'mode': 'PLAY', 'request_path': '/'}
2022-08-27 16:05:51.107 T:2783     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:64 Script params: {'item_id': '1ed8d400xxxxxxxxxx40c0fc503991e9', 'mode': 'PLAY', 'request_path': '/'}
2022-08-27 16:05:51.107 T:2783     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:140 JellyCon -> Mode: PLAY
2022-08-27 16:05:51.108 T:2783     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:141 JellyCon -> URL: None
2022-08-27 16:05:51.108 T:2783     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:804 == ENTER: PLAY ==
2022-08-27 16:05:51.108 T:2783     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:806 PLAY ACTION PARAMS: {'item_id': '1ed8d400xxxxxxxxxx40c0fc503991e9', 'mode': 'PLAY', 'request_path': '/'}
2022-08-27 16:05:51.108 T:2783     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:817 AUTO_RESUME: -1
2022-08-27 16:05:51.109 T:2783     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:820 FORCE_TRANSCODE: False
2022-08-27 16:05:51.109 T:2783     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:823 media_source_id: 
2022-08-27 16:05:51.109 T:2783     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:826 subtitle_stream_index: None
2022-08-27 16:05:51.110 T:2783     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:829 audio_stream_index: None
2022-08-27 16:05:51.114 T:2783     INFO <general>: JELLYFIN.resources.lib.functions -> INFO::resources/lib/functions.py:847 Sending jellycon_play_action : {'action': 'play', 'item_id': '1ed8d400xxxxxxxxxx40c0fc503991e9', 'auto_resume': '-1', 'force_transcode': False, 'media_source_id': '', 'subtitle_stream_index': None, 'audio_stream_index': None}
2022-08-27 16:05:51.115 T:2783     INFO <general>: JELLYFIN.resources.lib.play_utils -> DEBUG::resources/lib/play_utils.py:241 playFile id(1ed8d400xxxxxxxxxx40c0fc503991e9) resume(-1) force_transcode(False)
2022-08-27 16:05:51.156 T:2783     INFO <general>: JELLYFIN.resources.lib.play_utils -> DEBUG::resources/lib/play_utils.py:252 Playfile item: {'Name': 'MyChannel (DVB)', 'ServerId': '2d95109dxxxxxxxxxx0d5764c0f71050', 'Id': '1ed8d400xxxxxxxxxx40c0fc503991e9', 'Etag': '97d72608cxxxxxxxxxxc9317792bbc79', 'DateCreated': '2022-08-23T17:32:14.206Z', 'CanDelete': False, 'CanDownload': False, 'PreferredMetadataLanguage': '', 'PreferredMetadataCountryCode': '', 'SortName': '00001.0-MyChannel (DVB)', 'ForcedSortName': '', 'ExternalUrls': [], 'MediaSources': [{'Protocol': 'File', 'Id': '1ed8d400xxxxxxxxxx40c0fc503991e9', 'Type': 'Placeholder', 'Name': 'MyChannel (DVB)', 'IsRemote': False, 'ReadAtNativeFramerate': False, 'IgnoreDts': False, 'IgnoreIndex': False, 'GenPtsInput': False, 'SupportsTranscoding': True, 'SupportsDirectStream': True, 'SupportsDirectPlay': True, 'IsInfiniteStream': True, 'RequiresOpening': False, 'RequiresClosing': False, 'RequiresLooping': False, 'SupportsProbing': True, 'MediaStreams': [], 'MediaAttachments': [], 'Formats': [], 'RequiredHttpHeaders': {}}], 'EnableMediaSourceDisplay': True, 'CustomRating': '', 'ChannelId': None, 'Overview': '', 'Taglines': [], 'Genres': [], 'PlayAccess': 'Full', 'Number': '1', 'ChannelNumber': '1', 'RemoteTrailers': [], 'ProviderIds': {'ExternalServiceId': 'TVHclient LiveTvService'}, 'IsFolder': False, 'ParentId': 'b26028d1xxxxxxxxxx2f1bfc13f523d2', 'Type': 'TvChannel', 'People': [], 'Studios': [], 'GenreItems': [], 'LocalTrailerCount': 0, 'UserData': {'PlaybackPositionTicks': 0, 'PlayCount': 6, 'IsFavorite': False, 'LastPlayedDate': '2022-08-26T16:51:01.8218474Z', 'Played': False, 'Key': '1ed8d400-xxxx-xxxx-7c40-c0fc503991e9'}, 'SpecialFeatureCount': 0, 'DisplayPreferencesId': 'b5f10d1xxxxxxxxxxb2d3b2fda19134d', 'Tags': [], 'PrimaryImageAspectRatio': 1.6666666666666667, 'MediaStreams': [], 'ImageTags': {'Primary': '2a0b2ceexxxxxxxxxx5a14977b1d7032'}, 'BackdropImageTags': [], 'ImageBlurHashes': {}, 'LocationType': 'Remote', 'MediaType': 'Video', 'LockedFields': [], 'LockData': False, 'ChannelType': 'TV', 'CurrentProgram': {'Name': 'Best Show Ever', 'ServerId': '2d95109dxxxxxxxxxx0d5764c0f71050', 'Id': '406b7f107axxxxxxxxxx434e70a4ea10', 'DateCreated': '2022-08-23T17:32:43.1437869Z', 'SortName': 'best show ever', 'ExternalUrls': [], 'EnableMediaSourceDisplay': True, 'ChannelId': '1ed8d400xxxxxxxxxx40c0fc503991e9', 'ChannelName': 'MyChannel (DVB)', 'Overview': "This is really the best show ever.", 'Taglines': [], 'Genres': [], 'RunTimeTicks': 17999998976, 'PlayAccess': 'Full', 'ChannelNumber': '1', 'RemoteTrailers': [], 'ProviderIds': {}, 'ParentId': '1ed8d400xxxxxxxxxx40c0fc503991e9', 'Type': 'Program', 'People': [], 'Studios': [], 'GenreItems': [], 'LocalTrailerCount': 0, 'UserData': {'PlaybackPositionTicks': 0, 'PlayCount': 0, 'IsFavorite': False, 'Played': False, 'Key': '406b7f10-xxxx-xxxx-9866-434e70a4ea10'}, 'SpecialFeatureCount': 0, 'Tags': [], 'PrimaryImageAspectRatio': 1.7776, 'ImageTags': {'Primary': '93a7c507xxxxxxxxxx634a83d41f3dc0'}, 'BackdropImageTags': [], 'ImageBlurHashes': {}, 'MediaType': 'Video', 'EndDate': '2022-08-27T13:20:00.0000000Z', 'LockedFields': [], 'LockData': False, 'ChannelPrimaryImageTag': '2a0b2ceexxxxxxxxxx5a14977b1d7032', 'StartDate': '2022-08-27T12:50:00.0000000Z'}}
2022-08-27 16:05:51.189 T:2783     INFO <general>: JELLYFIN.resources.lib.play_utils -> DEBUG::resources/lib/play_utils.py:1631 PlaybackInfo : /Items/1ed8d400xxxxxxxxxx40c0fc503991e9/PlaybackInfo?MaxStreamingBitrate=75000000
2022-08-27 16:05:51.190 T:2783     INFO <general>: JELLYFIN.resources.lib.play_utils -> DEBUG::resources/lib/play_utils.py:1632 PlaybackInfo : {'Name': 'Kodi', 'MaxStaticBitrate': 75000000, 'MaxStreamingBitrate': 75000000, 'MusicStreamingTranscodingBitrate': 256000, 'TimelineOffsetSeconds': 5, 'TranscodingProfiles': [{'Type': 'Audio'}, {'Container': 'ts', 'Protocol': 'hls', 'Type': 'Video', 'AudioCodec': 'ac3', 'VideoCodec': 'h264', 'MaxAudioChannels': '8'}, {'Container': 'jpeg', 'Type': 'Photo'}], 'DirectPlayProfiles': [{'Type': 'Video'}, {'Type': 'Audio'}, {'Type': 'Photo'}], 'ResponseProfiles': [], 'ContainerProfiles': [], 'CodecProfiles': [], 'SubtitleProfiles': [{'Format': 'srt', 'Method': 'External'}, {'Format': 'srt', 'Method': 'Embed'}, {'Format': 'ass', 'Method': 'External'}, {'Format': 'ass', 'Method': 'Embed'}, {'Format': 'sub', 'Method': 'Embed'}, {'Format': 'sub', 'Method': 'External'}, {'Format': 'ssa', 'Method': 'Embed'}, {'Format': 'ssa', 'Method': 'External'}, {'Format': 'smi', 'Method': 'Embed'}, {'Format': 'smi', 'Method': 'External'}, {'Format': 'pgssub', 'Method': 'Embed'}, {'Format': 'pgssub', 'Method': 'External'}, {'Format': 'dvdsub', 'Method': 'Embed'}, {'Format': 'dvdsub', 'Method': 'External'}, {'Format': 'pgs', 'Method': 'Embed'}, {'Format': 'pgs', 'Method': 'External'}]}
2022-08-27 16:05:51.217 T:2783     INFO <general>: JELLYFIN.resources.lib.play_utils -> DEBUG::resources/lib/play_utils.py:1634 PlaybackInfo : {'MediaSources': [{'Protocol': 'Http', 'Id': '269288209', 'Path': 'http://mytvheadend.local:9981/stream/channelid/26xxxxx09?ticket=de019019xxxxxxxxxx660d2cd96fca6a468316a7', 'Type': 'Default', 'IsRemote': False, 'ReadAtNativeFramerate': False, 'IgnoreDts': False, 'IgnoreIndex': False, 'GenPtsInput': False, 'SupportsTranscoding': True, 'SupportsDirectStream': True, 'SupportsDirectPlay': True, 'IsInfiniteStream': True, 'RequiresOpening': False, 'RequiresClosing': False, 'BufferMs': 1500, 'RequiresLooping': False, 'SupportsProbing': True, 'MediaStreams': [{'VideoRange': 'SDR', 'VideoRangeType': 'SDR', 'DisplayTitle': 'SDR', 'NalLengthSize': '0', 'IsInterlaced': True, 'IsDefault': False, 'IsForced': False, 'Type': 'Video', 'Index': -1, 'IsExternal': False, 'IsTextSubtitleStream': False, 'SupportsExternalStream': False}, {'DisplayTitle': '', 'IsInterlaced': False, 'IsDefault': False, 'IsForced': False, 'Type': 'Audio', 'Index': -1, 'IsExternal': False, 'IsTextSubtitleStream': False, 'SupportsExternalStream': False}], 'MediaAttachments': [], 'Formats': [], 'RequiredHttpHeaders': {}, 'DefaultAudioStreamIndex': -1}], 'PlaySessionId': '387c948dxxxxxxxxxxfacad5c11b5121'}
2022-08-27 16:05:51.217 T:2783     INFO <general>: JELLYFIN.resources.lib.play_utils -> DEBUG::resources/lib/play_utils.py:383 play_session_id: 387c948dxxxxxxxxxxfacad5c11b5121
2022-08-27 16:05:51.218 T:2783     INFO <general>: JELLYFIN.resources.lib.play_utils -> DEBUG::resources/lib/play_utils.py:1125 get_play_url - media_source: {0}
2022-08-27 16:05:51.222 T:2783    ERROR <general>: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                                    - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'KeyError'>
                                                   Error Contents: 'Container'
                                                   Traceback (most recent call last):
                                                     File "/storage/.kodi/addons/plugin.video.jellycon/default.py", line 18, in <module>
                                                       main_entry_point()
                                                     File "/storage/.kodi/addons/plugin.video.jellycon/resources/lib/tracking.py", line 23, in wrapper
                                                       value = func(*args, **kwargs)
                                                     File "/storage/.kodi/addons/plugin.video.jellycon/resources/lib/functions.py", line 146, in main_entry_point
                                                       play_action(params)
                                                     File "/storage/.kodi/addons/plugin.video.jellycon/resources/lib/functions.py", line 848, in play_action
                                                       play_file(play_info)
                                                     File "/storage/.kodi/addons/plugin.video.jellycon/resources/lib/tracking.py", line 23, in wrapper
                                                       value = func(*args, **kwargs)
                                                     File "/storage/.kodi/addons/plugin.video.jellycon/resources/lib/play_utils.py", line 384, in play_file
                                                       playurl, playback_type, listitem_props = get_play_url(selected_media_source, play_session_id, channel_id)
                                                     File "/storage/.kodi/addons/plugin.video.jellycon/resources/lib/play_utils.py", line 1146, in get_play_url
                                                       container = media_source["Container"]
                                                   KeyError: 'Container'
                                                   -->End of Python script error report<--
                                                   
2022-08-27 16:05:51.560 T:2783     INFO <general>: Python interpreter stopped

----

2022-08-27 16:23:23.400 T:1093     INFO <general>: initializing python engine.
2022-08-27 16:23:24.152 T:1093     INFO <general>: JELLYFIN.default -> DEBUG::default.py:16 About to enter mainEntryPoint()
2022-08-27 16:23:24.153 T:1093     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:46 ===== JellyCon START =====
2022-08-27 16:23:24.159 T:1093     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:57 Running Python: sys.version_info(major=3, minor=8, micro=9, releaselevel='final', serial=0)
2022-08-27 16:23:24.159 T:1093     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:58 Running JellyCon: 0.5.7+py3
2022-08-27 16:23:24.160 T:1093     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:59 Kodi BuildVersion: 19.4 (19.4.0) Git:19.4-Matrix
2022-08-27 16:23:24.160 T:1093     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:60 Kodi Version: 19
2022-08-27 16:23:24.160 T:1093     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:61 Script argument data: ['plugin://plugin.video.jellycon/', '5', '?mode=SHOW_ADDON_MENU&type=library_item&view_id=b26028d1xxxxxxxxxx2f1bfc13f523d2', 'resume:false']
2022-08-27 16:23:24.161 T:1093     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:317 Parameter string: ?mode=SHOW_ADDON_MENU&type=library_item&view_id=b26028d1xxxxxxxxxx2f1bfc13f523d2
2022-08-27 16:23:24.161 T:1093     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:318 Plugin Path string: plugin://plugin.video.jellycon/
2022-08-27 16:23:24.161 T:1093     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:326 JellyCon -> Detected parameters: {'mode': 'SHOW_ADDON_MENU', 'type': 'library_item', 'view_id': 'b26028d1xxxxxxxxxx2f1bfc13f523d2', 'request_path': '/'}
2022-08-27 16:23:24.162 T:1093     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:64 Script params: {'mode': 'SHOW_ADDON_MENU', 'type': 'library_item', 'view_id': 'b26028d1xxxxxxxxxx2f1bfc13f523d2', 'request_path': '/'}
2022-08-27 16:23:24.195 T:1093     INFO <general>: JELLYFIN.resources.lib.menu_functions -> DEBUG::resources/lib/menu_functions.py:1131 VIEW_INFO : {'Name': 'Live TV', 'ServerId': '2d95109dxxxxxxxxxx0d5764c0f71050', 'Id': 'b26028d1xxxxxxxxxx2f1bfc13f523d2', 'Etag': 'd5bda1fxxxxxxxxxx58e49b8acc17564', 'DateCreated': '2022-08-23T17:30:12.5383099Z', 'CanDelete': False, 'CanDownload': False, 'SortName': 'live tv', 'ForcedSortName': 'Live TV', 'ExternalUrls': [], 'Path': '/config/metadata/views/livetv', 'EnableMediaSourceDisplay': True, 'ChannelId': None, 'Taglines': [], 'Genres': [], 'PlayAccess': 'Full', 'RemoteTrailers': [], 'ProviderIds': {}, 'IsFolder': True, 'ParentId': None, 'Type': 'UserView', 'People': [], 'Studios': [], 'GenreItems': [], 'LocalTrailerCount': 0, 'UserData': {'PlaybackPositionTicks': 0, 'PlayCount': 0, 'IsFavorite': False, 'Played': False, 'Key': 'b26028d1-xxxx-xxxx-c82f-1bfc13f523d2'}, 'ChildCount': 2, 'SpecialFeatureCount': 0, 'DisplayPreferencesId': 'cb46bc72exxxxxxxxxx072de3a65b93a', 'Tags': [], 'CollectionType': 'livetv', 'ImageTags': {}, 'BackdropImageTags': [], 'ImageBlurHashes': {}, 'LocationType': 'FileSystem', 'LockedFields': [], 'LockData': False}
2022-08-27 16:23:24.221 T:1093     INFO <general>: JELLYFIN.resources.lib.functions -> DEBUG::resources/lib/functions.py:166 ===== JellyCon FINISHED =====
2022-08-27 16:23:24.221 T:1093     INFO <general>: CPythonInvoker(9, /storage/.kodi/addons/plugin.video.jellycon/default.py): script successfully run
2022-08-27 16:23:24.566 T:1093     INFO <general>: Python interpreter stopped
2022-08-27 16:23:27.680 T:998      INFO <general>: JELLYFIN.resources.lib.websocket_client -> DEBUG::resources/lib/websocket_client.py:281 Reconnecting WebSocket
2022-08-27 16:23:32.690 T:998      INFO <general>: JELLYFIN.resources.lib.websocket_client -> DEBUG::resources/lib/websocket_client.py:237 Error: Handshake status 404 Not Found
2022-08-27 16:23:38.648 T:910      INFO <general>: JELLYFIN.resources.lib.menu_functions -> DEBUG::resources/lib/menu_functions.py:1188 set_library_window_values Called forced=False
2022-08-27 16:23:38.701 T:910      INFO <general>: JELLYFIN.resources.lib.widgets -> DEBUG::resources/lib/widgets.py:65 set_background_image Called forced=False
2022-08-27 16:23:38.720 T:910      INFO <general>: JELLYFIN.resources.lib.widgets -> DEBUG::resources/lib/widgets.py:80 set_background_image: Need to load more backgrounds 0 - 0
2022-08-27 16:23:38.750 T:910      INFO <general>: JELLYFIN.resources.lib.widgets -> DEBUG::resources/lib/widgets.py:108 set_background_image: Loaded 0 more backgrounds
2022-08-27 16:23:52.691 T:998      INFO <general>: JELLYFIN.resources.lib.websocket_client -> DEBUG::resources/lib/websocket_client.py:281 Reconnecting WebSocket
2022-08-27 16:23:58.826 T:910      INFO <general>: JELLYFIN.resources.lib.menu_functions -> DEBUG::resources/lib/menu_functions.py:1188 set_library_window_values Called forced=False
2022-08-27 16:23:58.871 T:910      INFO <general>: JELLYFIN.resources.lib.widgets -> DEBUG::resources/lib/widgets.py:65 set_background_image Called forced=False
2022-08-27 16:23:58.893 T:910      INFO <general>: JELLYFIN.resources.lib.widgets -> DEBUG::resources/lib/widgets.py:80 set_background_image: Need to load more backgrounds 0 - 0
2022-08-27 16:23:58.925 T:910      INFO <general>: JELLYFIN.resources.lib.widgets -> DEBUG::resources/lib/widgets.py:108 set_background_image: Loaded 0 more backgrounds
2022-08-27 16:24:02.707 T:998      INFO <general>: JELLYFIN.resources.lib.websocket_client -> DEBUG::resources/lib/websocket_client.py:237 Error: Handshake status 404 Not Found
2022-08-27 16:24:19.051 T:910      INFO <general>: JELLYFIN.resources.lib.menu_functions -> DEBUG::resources/lib/menu_functions.py:1188 set_library_window_values Called forced=False
2022-08-27 16:24:19.094 T:910      INFO <general>: JELLYFIN.resources.lib.widgets -> DEBUG::resources/lib/widgets.py:65 set_background_image Called forced=False
2022-08-27 16:24:19.115 T:910      INFO <general>: JELLYFIN.resources.lib.widgets -> DEBUG::resources/lib/widgets.py:80 set_background_image: Need to load more backgrounds 0 - 0
2022-08-27 16:24:19.140 T:910      INFO <general>: JELLYFIN.resources.lib.widgets -> DEBUG::resources/lib/widgets.py:108 set_background_image: Loaded 0 more backgrounds
```

--- a/resources/lib/play_utils.py
+++ b/resources/lib/play_utils.py
@@ -738,13 +738,19 @@ def audio_subs_pref(url, list_item, media_source, item_id, audio_stream_index, s
         index = stream['Index']
 
         if 'Audio' in stream['Type']:
-            codec = stream['Codec']
+            codec = stream.get('Codec', None)
             channel_layout = stream.get('ChannelLayout', "")
 
-            try:
-                track = "%s - %s - %s %s" % (index, stream['Language'], codec, channel_layout)
-            except:
-                track = "%s - %s %s" % (index, codec, channel_layout)
+            if not codec:
+                # Probably tvheadend and has no other info
+                track = "%s - default" % (index)
+            else:
+                try:
+                    # Track includes language
+                    track = "%s - %s - %s %s" % (index, stream['Language'], codec, channel_layout)
+                except:
+                    # Track doesn't include language
+                    track = "%s - %s %s" % (index, codec, channel_layout)
 
             audio_streams_list[track] = index
             audio_streams.append(track)
@@ -1143,7 +1149,6 @@ def get_play_url(media_source, play_session_id, channel_id=None):
     can_direct_play = media_source["SupportsDirectPlay"]
     can_direct_stream = media_source["SupportsDirectStream"]
     can_transcode = media_source["SupportsTranscoding"]
-    container = media_source["Container"]
 
     playurl = None
     playback_type = None
@@ -1155,6 +1160,7 @@ def get_play_url(media_source, play_session_id, channel_id=None):
         direct_path = direct_path.strip()
 
         # handle DVD structure
+        container = media_source["Container"]
         if container == "dvd":
             direct_path = direct_path + "/VIDEO_TS/VIDEO_TS.IFO"
         elif container == "bluray":
@@ -1172,17 +1178,9 @@ def get_play_url(media_source, play_session_id, channel_id=None):
     # check if file can be direct streamed/played
     if (can_direct_stream or can_direct_play) and playurl is None:
         item_id = media_source.get('Id')
-        # We need to include the channel ID if this is a live stream
         if channel_id:
-            url_root = '{}/Videos/{}/stream'.format(server, channel_id)
-            play_params = {
-                'static': True,
-                'PlaySessionId': play_session_id,
-                'MediaSourceId': item_id,
-                'LiveStreamId': media_source.get('LiveStreamId')
-            }
-            play_param_string = urlencode(play_params)
-            playurl = '{}?{}'.format(url_root, play_param_string)
+            # live tv has to be transcoded by the server
+            playurl = None
         else:
             url_root = '{}/Videos/{}/stream'.format(server, item_id)
             play_params = {
@@ -1230,7 +1228,8 @@ def get_play_url(media_source, play_session_id, channel_id=None):
 
         # We need to include the channel ID if this is a live stream
         if channel_id:
-            transcode_params['LiveStreamId'] = media_source.get('LiveStreamId')
+            if media_source.get('LiveStreamId'):
+                transcode_params['LiveStreamId'] = media_source.get('LiveStreamId')
             transcode_path = urlencode(transcode_params)
             playurl = '{}/Videos/{}/master.m3u8?{}'.format(
                 server, channel_id, transcode_path)

